src/AWS4AuthRequest.jl

```diff
@@ -83,21 +83,22 @@ function sign_aws4!(method::String,
     end
 
     # Sort and lowercase() Headers to produce canonical form...
-    canonical_headers = ["$(lowercase(k)):$(strip(v))" for (k,v) in headers]
-    signed_headers = join(sort([lowercase(k) for (k,v) in headers]), ";")
+    canonical_headers = sort!(map(headers) do (k, v)
+        string(lowercase(k), ':', replace(strip(v), r" +" => " "))
```

Comment:
Should this be any white space or just spaces?

Reply:
```suggestion
        string(lowercase(k), ':', replace(strip(v), r"\s+" => " "))
```